### PR TITLE
Tests for EGU peripheral for nRF5x targets

### PR DESCRIPTION
**Pull request by aescolar**
Re-submit of https://github.com/zephyrproject-rtos/zephyr/pull/48548 

Now the nrf52_bsim includes the EGU models. So this test can be included and run in CI (or in real HW locally).
The EGU peripheral is a HW interrupt generator and a PPI task/event sink/source.
This test exercises it, and at the same time tests quite a bit the plumbing around interrupts and control of peripherals events & tasks.

Compared to the original PR:
* It has been rebased.
* Checkpatch issues have been fixed.
* Include paths updated, project name changed, left over comments removed.
* An extra commit with a manifest update has been added 
* Repetitive code refactored into loops, empty lines removed, and some minor other nits fixed as per @MarGasiorek request
* Test updated to use new Ztest API

### CODE REVIEW

**Comment by aescolar** [comment]
CC: @FilipZajdel  , @RobertGalatNordic  , @MarGasiorek

**Review by RobertGalatNordic — DISMISSED** [review]

File: tests/boards/nrf52_bsim/egu/prj.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CONFIG_ZTEST=y
+CONFIG_TEST=y
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/prj.conf:2]: TEST is selected by [ZTEST](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/testsuite/ztest/Kconfig)

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+
+	nrf_timer_task_trigger(TIMER_INSTANCE, NRF_TIMER_TASK_START);
+	k_busy_wait(EGU_DELAY_USEC);
+
+	zassert_equal(call_cnt_expected, event_triggered_flag.call_count,
+			"interrupt called unexpected number of times %d",
+			event_triggered_flag.call_count);
+}
+
+
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:407]: Minor: empty line

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+}
+
+static void test_clean_egu(void)
+{
+	memset(NRF_EGU5, 0, sizeof(*NRF_EGU5));
+}
+
+void test_main(void)
+{
+
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:480]: Minor: empty line
Please check all file to eliminate the same issues

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+#define TIMER_DELAY_TICKS 100
+#define EGU_DELAY_USEC 200
+
+struct timer_isr_context {
+	unsigned int egu_channel;
+};
+
+void test_egu_channels_count(void)
+{
+	zassert_equal(16, nrf_egu_channel_count(NRF_EGU0), "NRF_EGU0 incorrect number of channels");
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:30]: Here I see duplicated code, did you consider make this part more generic?
My proposal (in pseudo code):
array  [NRF_EGU0 - NRF_EGU5]
and loop with asserts validation
for(test_egu_cnt=NRF_EGU0 to test_egu_cnt=NRF_EGU5)
{
zassert_equal(16, nrf_egu_channel_count(test_egu_cnt), "NRF_EGU%d incorrect number of channels", test_egu_cnt);
}

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+			  "NRF_EGU_5 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU5->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU5,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_5 incorrect address of event trigger 15");
+}
+
+void test_nrf_egu_channel_int_get(void)
+{
+	zassert_equal(NRF_EGU_INT_TRIGGERED0, nrf_egu_channel_int_get(0),
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:173]: Similar like in line 30, this bunch of code can be reduced to one for-loop and control by change number of elements in array

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+
+	zassert_equal(call_cnt_expected, event_triggered_flag.call_count,
+			"interrupt called unexpected number of times %d",
+			event_triggered_flag.call_count);
+}
+
+
+static struct SWI_trigger_assert_parameter SWI4_event_triggered_flag;
+
+void SWI4_trigger_function(const void *param)
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:410]: Is it a static function?

File: tests/boards/nrf52_bsim/egu/src/main.c [hunk 7]
```diff
@@ -0,0 +1,505 @@
...
+{
+	SWI4_event_triggered_flag.call_count++;
```
Comment (MarGasiorek) [inline tests/boards/nrf52_bsim/egu/src/main.c:412]: I don't understand what is purpose of this incremental.

Comment (aescolar) [reply tests/boards/nrf52_bsim/egu/src/main.c]: None for this ISR. Removed.

**Review by rugeGerritsen — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Review by kruithofa — APPROVED** [review]
LGTM

**Review by RobertGalatNordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by MarGasiorek — APPROVED** [review]
lgtm

**Pull request merged**